src/python/grpcio/grpc/_server.py

```diff
@@ -959,6 +958,17 @@ def add_secure_port(self, address, server_credentials):
     def start(self):
         _start(self._state)
 
+    def wait_for_termination(self, timeout=None):
+        termination_event = threading.Event()
```

Comment:
Change to:

```
def wait_for_termination(self, timeout=None):
    return self._state.termination_event.wait(timeout=timeout)
```

Reply:
The `return` is also important, because it helps the user identify if timeout expired or not

Reply:
Good catch. This is my oversight.